src/services/dmr/DmrService.ts

```diff
@@ -171,6 +190,101 @@ export class DmrService {
+      }
+    } else if (remove) {
+      builder.add(this.createRemoveOperation(address));
+    } else {
+      this.createWriteAttributeOperations(builder, address, config, excludedAttributes);
+      this.composeWriteObjectOperations(builder, address, config);
+    }
+  }
+
+  private createAddOperation(address: string[], config: any, excludedAttributes: string[]): IDmrRequest {
+    let allowedObjects: string[] = ["indexing-properties", "string-keyed-table", "binary-keyed-table"];
+    let op: IDmrRequest = {
+      address: address,
+      operation: "add"
+    };
+
+    if (isNotNullOrUndefined(config)) {
+      // iterate properties of model object and append (key/value) properties to op object
+      angular.forEach(config, (value, key) => {
+        if (isNullOrUndefined(value)) {
+          return;
+        }
+
+        if (isObject(value)) {
+          // Only process allowed objects, these should be objects which are dmr attributes not children
+          if (allowedObjects.indexOf(key) > -1) {
+            op[key] = value;
+          }
+          return;
+        }
+
+        if (isJsonString(value)) { // handle JSON described objects
+          value = this.parseJson(value, key);
+        }
+        op[key] = value;
+      });
+    }
+    // Remove excluded attributes
+    angular.forEach(excludedAttributes, attribute => delete op[attribute]);
+    return op;
+  }
+
+  private createRemoveOperation(address: string[]): IDmrRequest {
+    return <IDmrRequest> {
+      address: address,
+      operation: "remove"
+    };
+  }
+
+  private createWriteAttributeOperations(builder: CompositeOpBuilder, address: string[], config: any,
+                                         excludedAttributes: string[]): void {
+    if (isNotNullOrUndefined(config)) {
+      angular.forEach(config, (value, key) => {
+        let excluded: boolean = excludedAttributes.some(attribute => key === attribute) || isNullOrUndefined(value) || isObject(value);
+        if (!excluded) {
+          if (isJsonString(value)) {
+            value = this.parseJson(value, key);
+          }
+          builder.add(createWriteAttrReq(address, key, value));
+        }
+      });
+    }
+  }
+
+  private composeWriteObjectOperations(builder: CompositeOpBuilder, address: string[], config: any): void {
+    if (isNotNullOrUndefined(config)) {
+      let includedAttributes: string[] = ["indexing-properties", "string-keyed-table", "binary-keyed-table"];
+      angular.forEach(config, (value, key) => {
+        let included: boolean = includedAttributes.some(attribute => key === attribute);
+        if (included && isObject(value)) {
+          builder.add(createWriteAttrReq(address, key, value));
+        }
+      });
+    }
+  }
+
+  private parseJson(value: string, key: string): string {
+    try {
+      return JSON.parse(value);
+    } catch (e) {
+      console.log("Invalid JSON value " + value + " for field " + key);
+    }
+  }
```

Comment:
@ryanemerson what I meant to do is to retire all the mumbo jumbo DMR code from CacheConfigService and use the new model saving from DMRService. I'll do that later after this initial transition as a part of regular refactoring.